### PR TITLE
[baremetal and friends] Drop unnecessary readiness probes

### PR DESCRIPTION
**Pull request by cybertron**
Now that we are using the loadbalanced endpoint for keepalived
healthchecks, it is not correct to be using the local apiserver for
readiness checks. Also, I don't think there's any need for readiness
checks on the monitor. Kubernetes isn't routing any traffic to this
pod, so its ready status is meaningless.

### CODE REVIEW

**Comment by cybertron** [comment]
This can probably wait until 4.7 opens up. It's not causing any problems that I know of, I just noticed it as I was doing some other work with keepalived.

**Comment by yboaron** [comment]
I think you can delete also delete the readiness probe for baremetal Keepalived in the bootstrap https://github.com/openshift/machine-config-operator/blob/master/manifests/baremetal/keepalived.yaml#L129 , isn't it ?

I'm not sure if there's a K8S service that abstract the coredns static pods (I think the answer is no), so if there's no such a service I guess we can delete also the readiness probes from coredns static pod in all the platforms: 
https://github.com/openshift/machine-config-operator/blob/master/templates/common/baremetal/files/baremetal-coredns.yaml#L66

**Comment by cybertron** [comment]
I extended this change to all platforms and included coredns in the latest revision.That gets rid of all the readinessProbes in the templates and manifests directory AFAICT.

**Comment by cybertron** [comment]
One of these days I will remember to run make update before pushing MCO patches...

**Comment by jcpowermac** [comment]
openshift-tests might not pass but installed correctly
/lgtm

**Comment by kikisdeliveryservice** [comment]
waiting on lgtm from @rgolangh

**Review by mandre — COMMENTED** [review]
Just a heads up that the cloud powering OpenStack CI is down for maintenance and it's expected that e2e-openstack fails.

Patch looks good to me.

**Comment by kikisdeliveryservice** [comment]
Do we need this ovn job passing for this pr?

**Comment by cybertron** [comment]
I don't think so. It looks like that runs on AWS, which wouldn't be affected by this patch.

**Comment by sjenning** [comment]
@cybertron what is holding this up? Other than a rebase.  I'm very much wanting the warning on my web console to go away :smile:

**Comment by cybertron** [comment]
It was waiting for Roy's lgtm, but given that it's not really a platform-specific change I think we can probably go ahead at this point. However, I see now that it is in merge conflict so I'll need to get that addressed today.

**Comment by sjenning** [comment]
huge failures across CI due to new docker hub rate limiting on busybox image pull
```
docker.io/library/busybox: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```

**Review by mandre — COMMENTED** [review]
/lgtm

**Comment by kikisdeliveryservice** [comment]
one more try, test seems red

/test e2e-ovirt

**Review by kikisdeliveryservice — APPROVED** [review]

**Comment by kikisdeliveryservice** [comment]
@rgolangh the ovirt job seems very flaky, any insight?

**Comment by mandre** [comment]
/retest
The ovirt job passed, can we merge this?

**Pull request merged**
